src/ipa_pc_de/mod.rs

```diff
@@ -171,7 +164,8 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
         )
     }
 
-    /// Atomic verify function
+    /// The "succinct" verifier of a proof produced by `fn open()`. 
+    /// Does all a full verifier does, except the correctness check of G_fin.
     fn succinct_verify<'a>(
         vk: &VerifierKey<G>,
         combined_commitment: &'a Commitment<G>,
```

Comment:
What is the reason for calling the inputs combined_commitment, combined_value?

Reply:
Line 197:
This is not the best way to compute the combined commitment, since you are recomputing z^(is) at each step.
Observe that z^(is) = z^((i-1)*s) * z^s, so you don't need to recompute the power at each step.
It would be more efficient to use the Horner procedure for evaluating a polynomial:
You need to compute C_0 + a*C_1 + ... + a^n * C_n, where a = z^s.
Initialize C = C_n
for i = n-1...0: C = a*C + C_i